source/material_model/rheology/elasticity.cc

```diff
@@ -306,6 +363,33 @@ namespace aspect
             const double dte = elastic_timestep();
             const double dt = this->get_timestep();
 
+            // The viscosity should be averaged if material averaging is applied.
+            // Here the averaging scheme "project to Q1 (only viscosity)"  is
+            // excluded, because there is no way to know the quadrature formula
+            // used for evaluation.
+            // TODO: find a way to include "project to Q1 (only viscosity)" as well.
+            std::vector<double> effective_creep_viscosities;
+            if (this->get_parameters().material_averaging != MaterialAveraging::none &&
+                this->get_parameters().material_averaging != MaterialAveraging::project_to_Q1 &&
+                this->get_parameters().material_averaging != MaterialAveraging::project_to_Q1_only_viscosity)
+              {
+                MaterialModelOutputs<dim> out_copy(out.n_evaluation_points(),
+                                                   this->introspection().n_compositional_fields);
+                out_copy.viscosities = out.viscosities;
+
+                const MaterialAveraging::AveragingOperation averaging_operation_for_viscosity =
+                  get_averaging_operation_for_viscosity(this->get_parameters().material_averaging);
+                MaterialAveraging::average(averaging_operation_for_viscosity,
+                                           in.current_cell,
+                                           this->introspection().quadratures.compositional_fields,
```

Comment:
I think not. The Q1 projection only requires the number of quadrature points equals the length of vectors in MaterialModelOutputs, and MaterialModelOutputs is always constructed with the correct length, otherwise there would already be problems.

Reply:
I was just wondering whether the projected viscosity would end up different between the reaction terms and the RHS term because of the quadrature difference. Anyway it doesn't matter at the moment.